lib/cfn-nag/custom_rules/AmazonMQBrokerUserPasswordRule.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require 'cfn-nag/violation'
+require 'cfn-nag/util/enforce_reference_parameter'
+require 'cfn-nag/util/enforce_string_or_dynamic_reference'
+require_relative 'base'
+
+class AmazonMQBrokerUserPasswordRule < BaseRule
+
+  def rule_text
+    'Amazon MQ Broker resource Users property should exist and its Password property value ' +
+    'should not show password in plain text, resolve an unsecure ssm string, or have a default value for parameter.'
+  end
+
+  def rule_type
+    Violation::FAILING_VIOLATION
+  end
+
+  def rule_id
+    'F52'
+  end
+
+  def check_password_property(cfn_model, user)
+    # Checks to make sure 'Users' property has the key 'Password' defined.
+    # Also check if the value for the 'Password' key is nil
+    if user.has_key? 'Password'
+      if insecure_parameter?(cfn_model, user['Password'])
+        true
+      elsif insecure_string_or_dynamic_reference?(cfn_model, user['Password'])
+        true
+      elsif user['Password'].nil?
+        true
+      end
+    else
+      true    
+    end
+  end
+
+  def check_user_property(user_property)
```

Comment:
i'd nuke this method.  the return values could be true or nil which is possibly confusing - but you can check the nil? inline where this is invoked imho

Reply:
make this private